test/tests/functional/pbs_mom_walltime.py

```diff
@@ -83,24 +83,24 @@ def test_hold_time_not_counted_in_walltime(self):
 
         a = {'Resource_List.ncpus': 1}
         J1 = Job(TEST_USER, attrs=a)
-        J1.set_sleep_time(30)
+        J1.set_sleep_time(60)
         jid1 = self.server.submit(J1)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
-        time.sleep(10)
+        time.sleep(11)
         self.server.expect(JOB, {'resources_used.walltime': 0}, op=GT, id=jid1,
                            extend='x')
 
         self.server.holdjob(jid1, USER_HOLD)
         self.server.rerunjob(jid1)
         self.server.expect(JOB, {'Hold_Types': 'u'}, jid1)
-        time.sleep(5)
+        time.sleep(15)
 
         self.server.rlsjob(jid1, USER_HOLD)
         self.server.expect(JOB, {ATTR_state: 'F'}, id=jid1, extend='x',
-                           offset=30)
+                           offset=60)
 
         # The sleep time should not be included in the used walltime
-        self.server.expect(JOB, {'resources_used.walltime': 30}, op=LE,
+        self.server.expect(JOB, {'resources_used.walltime': 65}, op=LE,
```

Comment:
I am not sure if +5s is sufficient. can we do following instead?
- let the job run for 15s
- put it on hold
- read the used walltime
- wait for 20s
- read the walltime and make sure it is still the same
- release the job and let it finish.
- verify that walltime is in the range of 60-70s.

Reply:
@anamikau ,
Walltime of job will not change in held state. Also, how much variance is acceptable for walltime needs to be discussed before increasing the range. 70 seconds of walltime on a job of 60 seconds looks problematic.

Reply:
@anamikau  

- When the job is in held state the attribuite "resources_used.walltime" is not listed in  job's status(qstat -f). 

th2-mom01:/home/zulekam # qstat -f | grep job_state
    job_state = H
th2-mom01:/home/zulekam # qstat -f | grep resources_used.walltime
th2-mom01:/home/zulekam #

- However, the accounting logs will have this information in "R" record(As we are holding the job and requeuing the job using qrerun in the test)

10/04/2018 01:21:25;R;28.th2-mom01;user=zulekam group=altair project=_pbs_project_default jobname=STDIN queue=workq ctime=1538641256 qtime=1538641256 etime=1538641256 start=1538641256 exec_host=th2-mom01/0 exec_vnode=(th2-mom01:ncpus=1) Resource_List.ncpus=1 Resource_List.nodect=1 Resource_List.place=pack Resource_List.select=1:ncpus=1 session=2648 end=1538641285 Exit_status=-11 resources_used.cpupercent=0 resources_used.cput=00:00:00 resources_used.mem=600kb resources_used.ncpus=1 resources_used.vmem=8064kb resources_used.walltime=00:00:29 run_count=1

- But as you suggested, after waiting for 20s before qrls, I cannot read the updated walltime as both qstat and accounting logs won't have information related to updated used_walltime.

Reply:
@Ashish-Anand 
 "Walltime of job will not change in held state" 
--> this is what we are testing right?

Also, how much variance is acceptable for walltime needs to be discussed before increasing the range. 70 seconds of walltime on a job of 60 seconds looks problematic."
-->  MoM polls for resource usage for cput, walltime, mem and ncpus depending on the min_check_poll and max_check_poll. min_check_poll by default is 10s.
Also start of a new job resets the polling for all of the jobs being managed by this MoM.
If you still think we need to discuss then please post a question in the forum.

Reply:
@Mahalty thanks for the clarification. If that is the case then what you have is right. I still think you need to update the range.